test/k8sT/Services.go

```diff
@@ -802,10 +799,10 @@ var _ = Describe("K8sServicesTest", func() {
 			pod, err := kubectl.GetCiliumPodOnNode(helpers.CiliumNamespace, helpers.K8s2)
 			Expect(err).Should(BeNil(), "Cannot determine cilium pod name")
 
-			res := kubectl.CiliumExec(pod, "cilium service list | grep "+k8s2IP+":"+httpHostPortStr+" | grep HostPort")
+			res := kubectl.CiliumExecMustSucceed(context.TODO(), pod, "cilium service list | grep "+k8s2IP+":"+httpHostPortStr+" | grep HostPort")
 			Expect(res.GetStdOut()).ShouldNot(BeEmpty(), "No HostPort entry for "+k8s2IP+":"+httpHostPortStr)
```

Comment:
We can maybe remove the checks on `.ShouldNot(BeEmpty(), ...)` when the commands end with `grep`? I think `grep` returns non-zero if it does not find the value, so the `MustSucceed` part should fail, if I understand correctly? (Or replace with e.g. `awk` if you want to have a zero return code to make sure the command was run on the pod?)

And then in that case pass the error message from the `Expect()` to the `kubectl.CiliumExecMustSucceed()` instead I suppose?

Same remark for the other `grep` below.

Reply:
I am loath to change the test itself. If it's okay I just changed it to `CiliumExecContext`.

Reply:
I think you're trying to just improve the current situation in this PR (which it is improving!) without biting off too much all at once, so I think it's reasonable not to go and fix all the tests at once in one big PR. That said it sounds like this is something that the current test code really should be improved to do. Can we file an issue with `help-wanted` / `good-first-issue` to follow up on this?

Reply:
Done #10976